### PR TITLE
Migrate `$gray-lighter` SASS color to `$grey-3`

### PR DESCRIPTION
**Pull request by lyzadanger**
`$grey-lighter` (deprecated, `#d3d3d3`) to `$grey-3` (`#dbdbdb`). The colors are close enough in value, and their usages subtle enough (borders, typically) that I do not believe the change to be user-discernible. But if you want to compare, a good example is the input border color in the share-annotations dialog.

Also remove some unused styles that reference this deprecated color variable.

Also: Temporarily reorder all definitions of grey shades in `variables.scss` so we can see them in value order.

Part of https://github.com/hypothesis/client/issues/1560

### CODE REVIEW

File: src/styles/sidebar/components/annotation-quote.scss [hunk 1]
```diff
@@ -16,7 +16,7 @@
 .annotation-quote__quote {
   @include var.font-normal;
 
-  border-left: 3px solid var.$gray-lighter;
+  border-left: 3px solid var.$gray-3;
```
Comment (robertknight) [inline src/styles/sidebar/components/annotation-quote.scss]: This doesn't compile. Should be `var.$grey-3`.

File: src/styles/sidebar/sidebar.scss [hunk 2]
```diff
@@ -115,51 +115,10 @@ hypothesis-app {
   }
 }
 
-// Elements in root template (viewer.html)
-// ---------------------------------------
-
-.sheet {
```
Comment (robertknight) [inline src/styles/sidebar/sidebar.scss]: The `FocusedModeHeader` component still references the `sheet` and `sheet--short` classes.

Comment (lyzadanger) [reply src/styles/sidebar/sidebar.scss]: Facepalm. Bad grep methodology.

**Comment by lyzadanger** [comment]
@robertknight Derp derp derp. How about now?

File: src/styles/sidebar/sidebar.scss [hunk 3]
```diff
@@ -128,15 +128,6 @@ hypothesis-app {
   position: relative;
   background-color: var.$body-background;
 
-  .nav-tabs {
```
Comment (lyzadanger) [inline src/styles/sidebar/sidebar.scss]: This, however, does not appear to be used.

**Pull request merged**